File: backend/src/main/java/com/tribly/dto/posts/request/PostRequest.java
package com.tribly.dto.posts.request;

import com.tribly.dto.common.request.WithVisibility;
import com.tribly.dto.common.response.MediaDto;
import com.tribly.dto.validation.ValidateSchema;
import com.tribly.enums.Status;
import com.tribly.enums.Visibility;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import java.time.Instant;
import org.eclipse.microprofile.openapi.annotations.media.Schema;
import org.jspecify.annotations.Nullable;

@Schema(description = "Post request")
@ValidateSchema
public record PostRequest(
    @Schema(description = "Post name", required = true) @NotBlank @Size(min = 1, max = 200)
        String name,
    @Schema(description = "Post description", required = true) MediaDto media,
    @Schema(description = "Post date/time", required = true) Instant dateTime,
    @Schema(description = "Post status", required = true) Status status,
    @Schema(description = "Visibility level", required = true) Visibility visibility,
    @Nullable @Schema(description = "Publication timestamp (for scheduled publishing)")
        Instant publishAt)
    implements WithVisibility {}

File: frontend/src/api/dto/postRequest.ts
import type { Instant } from './instant.ts'
import type { MediaDto } from './mediaDto.ts'
import type { Status } from './status.ts'
import type { Visibility } from './visibility.ts'

/**
 * Post request
 */
export interface PostRequest {
  /**
   * Post name
   * @minLength 1
   * @maxLength 200
   * @pattern \S
   */
  name: string
  /** Post description */
  media: MediaDto
  /** Post date/time */
  dateTime: Instant
  /** Post status */
  status: Status
  /** Visibility level */
  visibility: Visibility
  /** Publication timestamp (for scheduled publishing) */
  publishAt?: Instant
}
